src/librustc_driver/lib.rs

```diff
@@ -1504,7 +1504,7 @@ pub fn in_rustc_thread<F, R>(f: F) -> Result<R, Box<Any + Send>>
             let err = io::Error::last_os_error();
             error!("in_rustc_thread: error calling getrlimit: {}", err);
             true
-        } else if rlim.rlim_max < STACK_SIZE as libc::rlim_t {
+        } else if rlim.rlim_max <= STACK_SIZE as libc::rlim_t {
             true
```

Comment:
I think you're reading this wrong -- `rlim_max` is the hard limit, so if that's less than the stack size we want, then we know `setrlimit` can't succeed, and we have to give up and start a new thread.  This `true` is the value of the expression assigned to `spawn_thread`.

We probably could have a case like `STACK_SIZE <= rlim.rlim_cur`, in which case the current stack size is sufficient, so it would set `false` from this expression (for `spawn_thread`).  Though perhaps we do want to reduce the limit too, just for safety!  The only case that definitely doesn't need `setrlimit` is when they are equal.